src/Components/Incidents/index.js

```diff
@@ -53,10 +53,23 @@ class Incidents extends PureComponent {
     });
   };
 
+  isFilteredOut = (incident) => {
+    const { filters } = this.context;
+    if (filters.exclude) {
+      const excluded = filters.exclude
+        .split(',')
+        .some((filter) => incident.service.summary.includes(filter));
+      if (excluded) return false;
+    }
+    if (
+      filters.showOnlyOwnIncidents
+      && incident.last_status_change_by.summary !== this.context.currentUser.name
+    ) return false;
+    return true;
+  };
+
   render() {
```

Comment:
`render()` became really long and complex. I suggest refactoring it a bit by extracting more complex parts (like L107-L114) to separate render methods.